packages/wrangler/src/__tests__/jest.setup.ts

```diff
@@ -90,10 +89,12 @@ afterEach(() => {
 afterAll(() => msw.close());
 
 jest.mock("../cfetch/internal");
+(fetchDashboardScript as jest.Mock).mockImplementation(
+	jest.requireActual("../cfetch/internal")["fetchDashboardScript"]
+);
```

Comment:
What does this mock do? Does it prevent an actual http request from happening? If so, is that antithetical to msw?

Reply:
No this allows for the fetchDashboardScript to be unmocked so that MSW can be triggered, once the MSW is triggered with a faux response, I remock it inside the helper function in the test file to handle the functions return of the script string. 
[related](https://github.com/cloudflare/wrangler2/pull/2418/files/8b976aa3c438c83c7bcac2dbe6b94baa6ad83df8#r1050116379)
```ts
// Mock the actual return value that FormData would return
	(fetchDashboardScript as jest.Mock).mockImplementationOnce(() => {
		return mockResponse;
	});
```

Reply:
Per offline discussion -- this _unmocks_ `fetchDashboardScript` in general, and then to work around the lack of form-data support, it gets re-mocked after sending a live http request. So there _is_ an actual http request happening, and once formdata support is merged into msw, we can remove this logic.